frontend/src/pages/modelRegistry/screens/RegisterModel.tsx

```diff
@@ -0,0 +1,319 @@
+import React from 'react';
+import {
+    ActionGroup,
+    Alert,
+    AlertActionCloseButton,
+    Breadcrumb,
+    BreadcrumbItem,
+    Button,
+    Form,
+    FormGroup,
+    FormSection,
+    HelperText,
+    HelperTextItem,
+    InputGroupItem,
+    InputGroupText,
+    Radio,
+    Stack,
+    StackItem,
+    Text,
+    TextInput,
+    Title,
+} from '@patternfly/react-core';
+import { useParams } from 'react-router';
+import { Link } from 'react-router-dom';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import useRegistrModelData from './useRegistrModelData';
+
+const RegisterModel: React.FC = () => {
+    const [
+        {
+            modelRegistryName,
+            modelName,
+            modelDescription,
+            versionName,
+            versionDescription,
+            sourceModelFormat,
+            modelLocationType,
+            modelLocationEndpoint,
+            modelLocationBucket,
+            modelLocationRegion,
+            modelLocationPath,
+            modelLocationURI,
+        },
+        setData,
+        resetData,
+    ] = useRegistrModelData();
+    const [loading, setIsLoading] = React.useState(false);
+    const [error, setError] = React.useState<Error | undefined>(undefined);
+    const { modelRegistry: mrName } = useParams();
+
+    React.useEffect(() => {
+        setData('modelRegistryName', mrName || "");
+    }, [setData]);
+
+    const isDisabled =
+        !modelRegistryName ||
+        !modelName ||
+        !versionName ||
+        loading ||
+        (modelLocationType === 'URI' && !modelLocationURI) ||
+        (modelLocationType === 'Object storage' &&
+            (!modelLocationBucket || !modelLocationEndpoint || !modelLocationPath));
+
+    const handleSubmit = () => {
+        setIsLoading(true);
+        setError(undefined);
+        /* eslint-disable-next-line no-console */
+        console.log({
+            modelRegistryName,
+            modelName,
+            modelDescription,
+            versionName,
+            versionDescription,
+            sourceModelFormat,
+            modelLocationType,
+            modelLocationEndpoint,
+            modelLocationBucket,
+            modelLocationRegion,
+            modelLocationPath,
+            modelLocationURI,
+        });
```

Comment:
Can we either remove the `console.log` here or add a TODO comment to remove it in the next PR?

Also, since we'll merge this without the submit behavior and we have other folks looking at the shared cluster, do you mind throwing in an `alert("This functionality is not yet implemented")` or something here so it doesn't seem broken?